pkg/download/mode/mode.go

```diff
@@ -64,11 +69,16 @@ func NewFile(m Mode, downloadURL string) (*DownloadFile, error) {
 		}
 		return parseFile(bts)
 	}
+
+	if m == "" {
```

Comment:
let's have this check at the top of the function right after the `op` declaration